addons/casings/functions/fnc_firedEH.sqf

```diff
@@ -0,0 +1,67 @@
+/*
+ * Author: Rocko and esteldunedain
+ * Create empty casing on weapon fired. Called from an ammo fired EH.
+ *
+ * Arguments:
+ * None. Parameters inherited from EFUNC(common,firedEH)
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+//IGNORE_PRIVATE_WARNING ["_unit", "_weapon", "_muzzle", "_mode", "_ammo", "_magazine", "_projectile", "_vehicle", "_gunner", "_turret"];
+
+BEGIN_COUNTER(fnc_fireEH);
+
+private _unitPosATL = getposATL _unit;
+
+// Make far away units 10 times less likely to create casings
+if ((positionCameraToWorld [0,0,0]) vectorDistance _unitPosATL > 100 && {random 1 < 0.9}) exitWith {};
```

Comment:
It get's the cameraPositon instead of the unit's head. 
This will also be the View Position when viewing through a UAV for example.